runtime/src/main/cpp/Exceptions.h

```diff
@@ -67,4 +67,23 @@ void PrintThrowable(KRef);
 } // extern "C"
 #endif
 
+// It's not always safe to extract SourceInfo during unhandled exception termination.
+class ScopedDisallowSourceInfo {
+ public:
+  ScopedDisallowSourceInfo();
+  ~ScopedDisallowSourceInfo();
+
+  ScopedDisallowSourceInfo(const ScopedDisallowSourceInfo&) = delete;
+  ScopedDisallowSourceInfo(ScopedDisallowSourceInfo&&) = delete;
+  ScopedDisallowSourceInfo& operator=(const ScopedDisallowSourceInfo&) = delete;
+  ScopedDisallowSourceInfo& operator=(ScopedDisallowSourceInfo&&) = delete;
+
+  static bool IsActive();
+
+ private:
+  // Making a counter instead of a boolean, if this lock ever gets nested.
+  // Shouldn't have any significant performance penalty.
+  static THREAD_LOCAL_VARIABLE int_fast8_t activeCount;
```

Comment:
Btw, what is the motivation behind using `int_fast8_t`? Is it an optimization?

Reply:
> Is it an optimization?

Essentially, yes. I wanted to use the fastest integer available and found the `int_fast*_t` family, which does exactly that. I don't think there's any harm in using those (except an element of surprise on the first encounter).

Also, with [Apple libc](https://opensource.apple.com/source/Libc/Libc-1353.41.1/include/stdint.h.auto.html) they seem to map to a corresponding `int*_t`, but on [glibc](https://sourceware.org/git/?p=glibc.git;a=blob;f=stdlib/stdint.h;h=2df07e485b402f9328c51e6cd0cabb0cf5afd6c1;hb=HEAD#l55) they do depend on a machine word size.

Reply:
I see. Well, I don't think this is worth optimizing :)

Reply:
I agree. The cost of doing so isn't very big, though.

Reply:
If you only need the fastest counter then plain `int` may be faster (and more universal) than any int8 variant.

Reply:
Technically, `int_fastx_t` is the fastest integer of size at least `x`. So, it may as well be `int` even for 8 byte variant. And on [glibc](https://sourceware.org/git/?p=glibc.git;a=blob;f=stdlib/stdint.h;h=2df07e485b402f9328c51e6cd0cabb0cf5afd6c1;hb=HEAD#l55) that's pretty much what's happening for 16 and 32 bit variants. For 8 bit, however, it uses `signed char` and I have no idea why.